torch/_prims/context.py

```diff
@@ -207,10 +208,19 @@ def _is_func_unsupported_nvfuser(torch_function_mode, func, args, kwargs):
 
     supported_ops = NvfuserPrimOperatorSupport()
     call_function_nodes = filter(lambda n: n.op == "call_function", gm.graph.nodes)
-    any_unsupported = any(
+    # check for any unsupported ops
+    if any(
         not supported_ops.is_node_supported(None, node) for node in call_function_nodes
-    )
-    return any_unsupported
+    ):
+        return True
+
+    flat_args, _ = tree_flatten(args)
+    # TODO: allow cpu scalar once we plumb the support via python API
+    # check for unsupport cpu tensor
+    if any(isinstance(arg, torch.Tensor) and arg.is_cpu and arg.ndim != 0 for arg in flat_args):  # type: ignore[attr-defined]
```

Comment:
So this line marks the node as unsupported if any of the arguments is a non-scalar CPU tensor? How can mixing devices happen without an error?
For example, addition is used in the tests, which just errors out as expected:
```py
import torch
from torch._prims.context import TorchRefsMode
a = torch.randn(3, 3, device="cuda")
b = torch.randn(3, 3, device="cpu")
# raises RuntimeError: Tensor on device cpu is not on the expected device cuda:0!
with TorchRefsMode():
    out = a + b

b = torch.tensor(0.5)
# works
with TorchRefsMode():
    out = a + b
```

Reply:
Why do we need to resort to aten op when even there mixing cuda and cpu tensors should error out:
```py
In [1]: import torch

In [2]: a = torch.randn(3, 3, device='cuda')

In [3]: a + a.cpu()
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Input In [3], in <cell line: 1>()
----> 1 a + a.cpu()

RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
```

Reply:
The magic here is `arg.ndim != 0`.

mix device only really works with cpu scalar, in which case it's usually handled by cuda kernels that takes cpu scalar as a regular scalar.

```
In [1]: import torch

In [2]: a = torch.randn(3, 3, device="cuda")

In [3]: y = torch.tensor(0.5)

In [4]: y
Out[4]: tensor(0.5000)

In [5]: y.device
Out[5]: device(type='cpu')

In [6]: print(a + y)
tensor([[ 0.0930,  1.7815,  0.8127],
        [ 0.4306, -0.0681,  0.4698],
        [ 1.7451, -1.8241,  1.1352]], device='cuda:0')

```

Reply:
Yes I see `arg.ndim == 0` is allowed.

I mean why should we have any special login for the case of `arg.ndim !=0` if it errors out also in eager and primtorch?